sound/soc/sof/intel/hda-codec.c

```diff
@@ -154,12 +154,15 @@ int hda_codec_probe_bus(struct snd_sof_dev *sdev,
 
 		ret = hda_codec_probe(sdev, i, hda_codec_use_common_hdmi);
 		if (ret < 0) {
-			dev_err(bus->dev, "error: codec #%d probe error, ret: %d\n",
-				i, ret);
-			return ret;
+			dev_warn(bus->dev, "codec #%d probe error, ret: %d\n",
+				 i, ret);
+			bus->codec_mask &= ~BIT(i);
 		}
 	}
 
+	if (!bus->codec_mask)
+		return -ENODEV;
+
```

Comment:
/* create codec instances */
	hda_codec_probe_bus(sdev, hda_codec_use_common_hdmi);

so that -ENODEV would have no impact...

Reply:
@plbossart That matches the existing implementation, but you are right the errors (if passed) should be handled. I now updated the patch and made the probe function return void. I did also try a version that returns the error and passes it up from the caller, but we actually get more understandable error messages if we let this pass and get an error message from machine_check(). Who knows, maybe there's some case where DSP will be loaded despite no codecs found.